substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/resources/NativeImageResourceFileStore.java

```diff
@@ -0,0 +1,131 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.core.jdk.resources;
+
+import java.io.IOException;
+import java.nio.file.FileStore;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.BasicFileAttributeView;
+import java.nio.file.attribute.FileAttributeView;
+import java.nio.file.attribute.FileStoreAttributeView;
+
+public class NativeImageResourceFileStore extends FileStore {
+
+    private final FileSystem resourceFileSystem;
+
+    NativeImageResourceFileStore(Path path) {
+        this.resourceFileSystem = path.getFileSystem();
+    }
+
+    @Override
+    public String name() {
+        return resourceFileSystem.toString() + "/";
+    }
+
+    @Override
+    public String type() {
+        return "rfs";
+    }
+
+    @Override
+    public boolean isReadOnly() {
+        return resourceFileSystem.isReadOnly();
+    }
+
+    @Override
+    public long getTotalSpace() throws IOException {
+        return new ResourceFileStoreAttributes(this).totalSpace();
+    }
+
+    @Override
+    public long getUsableSpace() throws IOException {
+        return new ResourceFileStoreAttributes(this).usableSpace();
+    }
+
+    @Override
+    public long getUnallocatedSpace() throws IOException {
+        return new ResourceFileStoreAttributes(this).unallocatedSpace();
+    }
+
+    @Override
+    public boolean supportsFileAttributeView(Class<? extends FileAttributeView> type) {
+        return type == BasicFileAttributeView.class || type == NativeImageResourceFileAttributesView.class;
+    }
+
+    @Override
+    public boolean supportsFileAttributeView(String name) {
+        return name.equals("basic") || name.equals("resource");
+    }
+
+    @Override
+    public <V extends FileStoreAttributeView> V getFileStoreAttributeView(Class<V> type) {
+        if (type == null) {
+            throw new NullPointerException();
+        }
+        return null;
+    }
+
+    @Override
+    public Object getAttribute(String attribute) throws IOException {
+        if (attribute.equals("totalSpace")) {
+            return getTotalSpace();
+        }
+        if (attribute.equals("usableSpace")) {
+            return getUsableSpace();
+        }
+        if (attribute.equals("unallocatedSpace")) {
+            return getUnallocatedSpace();
+        }
+        throw new UnsupportedOperationException("Attribute isn't supported!");
+    }
+
+    private static class ResourceFileStoreAttributes {
+
+        private final FileStore fileStore;
+        private final long size;
+
+        ResourceFileStoreAttributes(NativeImageResourceFileStore fileStore) throws IOException {
+            Path path = FileSystems.getDefault().getPath(fileStore.name());
```

Comment:
If I understood correctly, a `NativeImageResourceFileStore` is bound to a path inside of the native-image file system. However, here we seem to be resolving that path in the default file system - I may have missed something, but it seems like this file isn't connected to the native-image file system and the information here, provided that such a path exists, most likely isn't what we're looking to return